pkg/config/vault_test.go

```diff
@@ -227,6 +228,90 @@ func addAlias(param *StepParameters, aliasName string) {
 	param.Aliases = append(param.Aliases, alias)
 }
 
+func TestResolveVaultTestCredentialsWrapper(t *testing.T) {
+	t.Parallel()
+	t.Run("Default test credential prefix", func(t *testing.T) {
+		t.Parallel()
+		// init
+		vaultMock := &mocks.VaultMock{}
+		envPrefix := "PIPER_TESTCREDENTIAL_"
+		stepConfig := StepConfig{Config: map[string]interface{}{
+			"vaultPath":               "team1",
+			"vaultTestCredentialPath": []interface{}{"appCredentials1", "appCredentials2"},
+			"vaultTestCredentialKeys": []interface{}{[]interface{}{"appUser1", "appUserPw1"}, []interface{}{"appUser2", "appUserPw2"}},
+		}}
+
+		defer os.Unsetenv("PIPER_TESTCREDENTIAL_APPUSER1")
+		defer os.Unsetenv("PIPER_TESTCREDENTIAL_APPUSERPW1")
+		defer os.Unsetenv("PIPER_TESTCREDENTIAL_APPUSER2")
+		defer os.Unsetenv("PIPER_TESTCREDENTIAL_APPUSERPW2")
+
+		// mock
+		vaultData1 := map[string]string{"appUser1": "test-user", "appUserPw1": "password1234"}
+		vaultMock.On("GetKvSecret", "team1/appCredentials1").Return(vaultData1, nil)
+		vaultData2 := map[string]string{"appUser2": "test-user", "appUserPw2": "password1234"}
+		vaultMock.On("GetKvSecret", "team1/appCredentials2").Return(vaultData2, nil)
+
+		// test
+		resolveVaultTestCredentialsWrapper(&stepConfig, vaultMock)
+
+		// assert
+		for k, expectedValue := range vaultData1 {
+			env := envPrefix + strings.ToUpper(k)
+			assert.NotEmpty(t, os.Getenv(env))
+			assert.Equal(t, expectedValue, os.Getenv(env))
+		}
+
+		// assert
+		for k, expectedValue := range vaultData2 {
+			env := envPrefix + strings.ToUpper(k)
+			assert.NotEmpty(t, os.Getenv(env))
+			assert.Equal(t, expectedValue, os.Getenv(env))
+		}
+	})
+
+	// Test empty and non-empty custom general purpose credential prefix
+	envPrefixes := []string{"CUSTOM_MYCRED1_", ""}
+	for idx, envPrefix := range envPrefixes {
+		tEnvPrefix := envPrefix
```

Comment:
What does `tEnvPrefix` do, compared to just using `envPrefix`?

Reply:
envPrefix is passed to t.Run() but it is running in parallel using t.Parallel(), so envPrefix is mutated on each iteration leaving us with a bug, the value of envPrefix will be the last element of envPrefixes. So in order to avoid it, we can just create a variable on each iteration and pass it to t.Run() which is running in parallel.